substrate/client/executor/polkavm/src/lib.rs

```diff
@@ -205,25 +188,25 @@ fn call_host_function(
 		Ok(value) => value,
 		Err(error) => {
 			log::warn!("Call into the host function '{}' failed: {error}", function.name());
-			return Err(polkavm::Trap::default());
+			return Err(polkavm::CallError::Trap);
 		},
 	};
 
 	if let Some(value) = value {
 		match value {
 			Value::I32(value) => {
-				caller.set_reg(Reg::A0, value as u32);
+				caller.instance.set_reg(Reg::A0, value as u64);
 			},
 			Value::F32(value) => {
-				caller.set_reg(Reg::A0, value);
+				caller.instance.set_reg(Reg::A0, value as u64);
 			},
 			Value::I64(value) => {
-				caller.set_reg(Reg::A0, value as u32);
-				caller.set_reg(Reg::A1, (value >> 32) as u32);
+				caller.instance.set_reg(Reg::A0, value as u64);
+				caller.instance.set_reg(Reg::A1, (value >> 32) as u64);
 			},
 			Value::F64(value) => {
-				caller.set_reg(Reg::A0, value as u32);
-				caller.set_reg(Reg::A1, (value >> 32) as u32);
+				caller.instance.set_reg(Reg::A0, value as u64);
+				caller.instance.set_reg(Reg::A1, (value >> 32) as u64);
```

Comment:
Same as previous comment; check bitness and handle accordingly.

Reply:
OK here this is clear as set_reg exists.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/6533/commits/def26a5994f673ba4a8310cdf61068f9aab251af